akka-actor-typed/src/main/scala/akka/actor/typed/internal/routing/PoolRouterImpl.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.internal.routing
+
+import akka.actor.typed.Behavior.DeferredBehavior
+import akka.actor.typed._
+import akka.actor.typed.scaladsl.AbstractBehavior
+import akka.actor.typed.scaladsl.ActorContext
+import akka.actor.typed.scaladsl.Behaviors
+import akka.annotation.InternalApi
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] final case class PoolRouterBuilder[T](
+  poolSize:     Int,
+  behavior:     Behavior[T],
+  logicFactory: () ⇒ RoutingLogic[T] = () ⇒ new RoutingLogics.RoundRobinLogic[T]
+) extends DeferredBehavior[T]
+  with scaladsl.PoolRouter[T]
+  with javadsl.PoolRouter[T] {
+
+  // deferred creation of the actual router
+  def apply(ctx: TypedActorContext[T]): Behavior[T] = PoolRouterImpl[T](poolSize, behavior, logicFactory)
+
+  def withRandomRouting(): PoolRouterBuilder[T] = copy(logicFactory = RoutingLogics.randomLogic[T])
+
+  def withRoundRobinRouting(): PoolRouterBuilder[T] = copy(logicFactory = () ⇒ new RoutingLogics.RoundRobinLogic[T])
+}
```

Comment:
good, if we ever open up for custom routing logic it can be added here; `withRoutingLogic`